numcodecs/registry.py

```diff
@@ -16,7 +16,7 @@ def run_entrypoints():
         entries.update({e.name: e for e in eps.select(group="numcodecs.codecs")})
     else:
         # Otherwise, fallback to using get
-        entries.update(eps.get("numcodecs.codecs", []))
+        entries.update(eps.get("numcodecs.codecs", {}))
```

Comment:
Looks like this particular line traces back to PR: https://github.com/zarr-developers/numcodecs/pull/442

cc @s-t-e-v-e-n-k (in case you have thoughts on this change)

Reply:
I guess so? I will admit I was thinking about what `entry_points().get()` would return, rather than updating the dictionary. I would feel more comfortable if there was a test case with no entry points, but I apologise for not adding a test case myself.

Reply:
I'm trying to add a test now and I'm no longer certain I've isolated quite the right case that is triggering the issue. Stay tuned.

Reply:
OK, I've written up my full post-portem in #478 now. The problem is far more esoteric than I initially diagnosed. I'm not sure a test here is going to be very feasible, please have a look at my writeup in the issue and let me know what you think.

Reply:
All right, I think I've found an OK way to write a test. Let's see if it really isolates well enough to avoid mucking with other tests though.

Reply:
I will admit I'm not happy with how importlib_metadata plays with importlib.metadata, and that I didn't consider API breaking changes between them, since I was only testing with importlib.metadata, but other projects might utilise numcodecs and something else that requires importlib_metadata. :-(

Reply:
Yeah this is a pretty gross situation to wind up in. I haven't had to dive this deep in before and it took me a couple hours to really zero in on what was going on. Totally understandable how this could get missed in numcodecs. It feels like a problem that isn't really going to get fixed until Python 3.9 EOLs :disappointed: